src/firefly/js/ui/DownloadDialog.jsx

```diff
@@ -153,8 +160,17 @@ export function DownloadOptionPanel (props) {
     };
     const maskPanel = <BgMaskPanel componentKey={bgKey} style={maskStyle}/>;
 
+    const saveAsProps = {
+        fieldKey: 'BaseFileName',
+        initialState: {
+            value: get(dlParams, 'BaseFileName')
+        }
+    };
+
+    const dlTitle = get(dlParams, 'TitlePrefix', 'Download') + '-' + dlTitleIdx;
```

Comment:
Although the index is increased, in my test, the title stays the same on each submit.